workflow-manager/src/domain/cosmos_cw.rs

```diff
@@ -652,6 +676,148 @@ impl Connector for CosmosCosmwasmConnector {
 
         Ok(())
     }
+
+    async fn query_workflow_registry(
+        &mut self,
+        main_domain: &str,
+        id: u64,
+    ) -> ConnectorResult<valence_workflow_registry_utils::WorkflowResponse> {
+        if self.chain_name != main_domain {
+            return Err(CosmosCosmwasmError::Error(anyhow::anyhow!(
+                "Adding external domain is only possible on main domain in authorization contract"
+            ))
+            .into());
+        }
+
+        // Query for workflow config for an idea
+        let query_data = to_vec(&valence_workflow_registry_utils::QueryMsg::GetConfig { id })
+            .map_err(CosmosCosmwasmError::SerdeJsonError)?;
+        let config_req = QuerySmartContractStateRequest {
+            address: CONFIG.get_registry_addr().clone(),
+            query_data,
+        };
+
+        let config_res = from_json::<valence_workflow_registry_utils::WorkflowResponse>(
+            self.wallet
+                .client
+                .clients
+                .wasm
+                .smart_contract_state(config_req)
+                .await
+                .context("'query_workflow_registry' Failed to query workflow registry")
+                .map_err(CosmosCosmwasmError::Error)?
+                .into_inner()
+                .data,
+        )
+        .map_err(CosmosCosmwasmError::CosmwasmStdError)?;
+
+        Ok(config_res)
+    }
+
+    async fn verify_account(&mut self, account_addr: String) -> ConnectorResult<()> {
+        let contract_name = self.get_contract_name_by_code_id(account_addr).await?;
+
+        // Loop over account types and see if any matches the contract name of the code id
+        // error if it doesn't match, else return ()
+        Ok(AccountType::VARIANTS
+            .iter()
+            .find(|x| x.to_string() == contract_name)
+            .context("'verify_account' Code id doesn't match any account type")
+            .map(|_| ())
+            .map_err(CosmosCosmwasmError::Error)?)
+    }
+
+    async fn verify_service(&mut self, service_addr: Option<String>) -> ConnectorResult<()> {
+        let service_addr = service_addr
+            .context("'verify_service' Service address is empty")
+            .map_err(CosmosCosmwasmError::Error)?;
+
+        let contract_name = self.get_contract_name_by_code_id(service_addr).await?;
+
+        Ok(ServiceConfig::VARIANTS
+            .iter()
+            .find(|x| x.to_string() == contract_name)
+            .context("'verify_account' Code id doesn't match any account type")
+            .map(|_| ())
+            .map_err(CosmosCosmwasmError::Error)?)
+    }
+
+    async fn verify_processor(&mut self, processor_addr: String) -> ConnectorResult<()> {
+        let contract_name = self.get_contract_name_by_code_id(processor_addr).await?;
+
+        // Make sure the code id is of name processor
+        if contract_name != "processor" {
```

Comment:
nit: should we use valence-processor, to stay consistent on contracts/workflow manager?

Reply:
I would like not to, but I can see why that might be preferred, we already remove valence prefix in our folder names for example.
The reason I prefer not to, is because this naming is not just for the config, rather its also used in enums for example, keep it as simple as possible is best in this case.

@stiiifff @bekauz Thoughts?

Reply:
This seems a bit weak in terms of verification .. maybe query the contract's cw2 info (and verify the contract's crate name) would be better ?